message/validation/consensus_state.go

```diff
@@ -4,36 +4,29 @@ import (
 	"sync"
 
 	"github.com/attestantio/go-eth2-client/spec/phase0"
-	spectypes "github.com/ssvlabs/ssv-spec/types"
 )
 
-// consensusID uniquely identifies a public key and role pair to keep track of state.
-type consensusID struct {
-	DutyExecutorID string
-	Role           spectypes.RunnerRole
-}
-
-// consensusState keeps track of the signers for a given public key and role.
-type consensusState struct {
-	state           map[spectypes.OperatorID]*OperatorState
+// ValidatorState keeps track of the signers for a given public key and role.
+type ValidatorState struct {
+	operators       []*OperatorState
 	storedSlotCount phase0.Slot
 	mu              sync.Mutex
 }
 
-func (cs *consensusState) GetOrCreate(signer spectypes.OperatorID) *OperatorState {
+func (cs *ValidatorState) Signer(idx int) *OperatorState {
 	cs.mu.Lock()
 	defer cs.mu.Unlock()
 
-	if _, ok := cs.state[signer]; !ok {
-		cs.state[signer] = newOperatorState(cs.storedSlotCount)
+	if cs.operators[idx] == nil {
+		cs.operators[idx] = newOperatorState(cs.storedSlotCount)
 	}
 
-	return cs.state[signer]
+	return cs.operators[idx]
 }
 
 type OperatorState struct {
-	mu              sync.RWMutex
-	state           []*SignerState // the slice index is slot % storedSlotCount
+	mu              sync.Mutex
+	signers         []*SignerState // the slice index is slot % storedSlotCount
```

Comment:
it's operator/signer state indexed by slot, so I'm not sure if this name is better